plugins/ftdidmx/FtdiDmxThread.cpp

```diff
@@ -73,15 +97,133 @@ bool FtdiDmxThread::WriteDMX(const DmxBuffer &buffer) {
   }
 }
 
+void FtdiDmxThread::SendRDMRequest(ola::rdm::RDMRequest *request,
+                    ola::rdm::RDMCallback *callback) {
+  ola::thread::MutexLocker locker(&m_rdm_mutex);
+  if(m_pending_request == nullptr) {
+      m_pending_request = request;
+      m_rdm_callback = callback;
+  } else {
+    OLA_WARN << "Unable to queue RDM request, RDM operation already pending";
+  }
+  OLA_WARN << "Function not properly implemented yet, callback may not get called.";
+}
+
+void FtdiDmxThread::RunFullDiscovery(ola::rdm::RDMDiscoveryCallback *callback) {
+  m_discovery_agent.StartFullDiscovery(ola::NewSingleCallback(this, &FtdiDmxThread::DiscoveryComplete, callback));
+}
+
+void FtdiDmxThread::RunIncrementalDiscovery(ola::rdm::RDMDiscoveryCallback *callback) {
+  m_discovery_agent.StartIncrementalDiscovery(ola::NewSingleCallback(this, &FtdiDmxThread::DiscoveryComplete, callback));
+}
+
+/**
+ * Called when the discovery process finally completes
+ * @param callback the callback passed to StartFullDiscovery or
+ * StartIncrementalDiscovery that we should execute.
+ * @param status true if discovery worked, false otherwise
+ * @param uids the UIDSet of UIDs that were found.
+ */
+void FtdiDmxThread::DiscoveryComplete(ola::rdm::RDMDiscoveryCallback *callback,
+                                       bool,
+                                       const ola::rdm::UIDSet &uids) {
+  OLA_DEBUG << "FTDI discovery complete: " << uids;
+  if (callback) {
+    callback->Run(uids);
+  }
+}
+
+/**
+ * @brief Method called to cleanup any outstanding callbacks
+ * @param state
+ *
+ * All callbacks except the RDMCallback lack a way of reporting an error state to the caller.
```

Comment:
@note

Reply:
?

Reply:
Sorry, I meant use the @note form of Doxygen flag, so it doesn't just disappear into the description.